spec/src/main/asciidoc/core/implementation.asciidoc

```diff
@@ -743,7 +743,7 @@ Certain legal bean types cannot be proxied by the container:
 
 * classes which don't have a non-private constructor with no parameters,
 * classes which are declared final,
-* classes which have non-static, final methods with public, protected or default visibility,
+* classes which have non-static, final methods with public, protected or default visibility except the classes are explicitly enlisted via <<allowProxying>>,
```

Comment:
Definitely not for final classes. For non-private constructors it's possible using non-portable JVM APIs.